InformationScripting/src/queries/RuntimeQuery.h

```diff
@@ -41,7 +43,9 @@ class INFORMATIONSCRIPTING_API RuntimeQuery : public LinearQuery
 	private:
 		friend class QueryRegistry;
 
-		RuntimeQuery(Model::Node* target, QStringList);
+		QueryExecutor* executor_{};
+
+		RuntimeQuery(Model::Node* target, QStringList, QueryExecutor* executor = nullptr);
```

Comment:
I see. Well you don't technically need `static_if`. That would be a direct way of doing things, but one could use explicit specializations to achieve the same effect.

In any case, you don't have to do it as it will be quite some additional code.